src/Microsoft.AspNet.Diagnostics.Elm/Views/DetailsPage.cshtml

```diff
@@ -14,56 +15,86 @@
     }
 
     public DetailsPageModel Model { get; set; }
-}
-
-@helper LogRow(LogInfo log)
-{
-    if (log.Severity >= Model.Options.MinLevel &&
-        (string.IsNullOrEmpty(Model.Options.NamePrefix) || log.Name.StartsWith(Model.Options.NamePrefix, StringComparison.Ordinal)))
-    {
-        <tr>
-            <td>@string.Format("{0:MM/dd/yy}", log.Time)</td>
-            <td>@string.Format("{0:H:mm:ss}", log.Time)</td>
-            <td class="@log.Severity.ToString().ToLowerInvariant()">@log.Severity</td>
-            <td title="@log.Name">@log.Name</td>
-            <td title="@log.Message" class="logState" width="100px">@log.Message</td>
-            <td title="@log.Exception">@log.Exception</td>
-        </tr>
-    }
-}
 
-@helper Traverse(ScopeNode node)
-{
-    var messageIndex = 0;
-    var childIndex = 0;
-    while (messageIndex < node.Messages.Count && childIndex < node.Children.Count)
-    {
-        if (node.Messages[messageIndex].Time < node.Children[childIndex].StartTime)
-        {
-            @LogRow(node.Messages[messageIndex])
-            messageIndex++;
-        }
-        else
-        {
-            @Traverse(node.Children[childIndex])
-            childIndex++;
-        }
-    }
-    if (messageIndex < node.Messages.Count)
+    public HelperResult LogRow(LogInfo log)
     {
-        for (var i = messageIndex; i < node.Messages.Count; i++)
+        return new HelperResult((writer) =>
         {
-            @LogRow(node.Messages[i])
-        }
+            if (log.Severity >= Model.Options.MinLevel &&
+               (string.IsNullOrEmpty(Model.Options.NamePrefix) || log.Name.StartsWith(Model.Options.NamePrefix, StringComparison.Ordinal)))
+            {
+                WriteLiteralTo(writer, "        <tr>\r\n            <td>");
+                WriteTo(writer, string.Format("{0:MM/dd/yy}", log.Time));
+                WriteLiteralTo(writer, "</td>\r\n            <td>");
+                WriteTo(writer, string.Format("{0:H:mm:ss}", log.Time));
+                WriteLiteralTo(writer, "</td>\r\n            <td");
+                WriteAttributeTo(writer, "class", " class=\"", "\"",
```

Comment:
I'd kind of like to keep the whitespace. Makes the code here ugly, but better for society :smile: